docs/Current/primary-cloud-for-web-applications.md

```diff
@@ -0,0 +1,147 @@
+### Alt 3: GCP-only strategy (for web apps)
+In this solution we use GCP as the primary platform for hosting web applications. In this scenario, Azure could still be used as a user directory and synchronized with GCP.
+
+GCP is developed by Google and in _general_ the following can be said of GCP:
+1. Low cost for realtime db (propriotory firebase)
+2. Good performance and low costs associated with hosting APIs (Cloud run)
+3. Good (better) security configuration "by default" (esp. things like cloudproxy)
+4. Pricing often slightly lower than Azure (with some notable exceptions like BigQuery, SpannerSQL).
+5. Monitoring and tracing tooling is less mature and more "clunky" than on Azure, but supports Go-lang out of the box.
+
+**Risks**  
+1. Need to migrate existing projects from Azure - many of which run on .Net (poorer experience)
+2. Smaller offering and missing features compared to Azure
+3. No presence in Norway (yet)
+4. Poorer monitoring experience
+5. Need to manage security, costs etc. in multiple clouds (due to corporate applications)
+
+**Benefits**  
+1. In general, cheaper and robust services (e.g. cloud run)
+2. Loved by existing teams
+
+### Alt 4: Low Cost hybrid strategy
+There are a number of alternative cloud providers such as Linode, Digital Ocean etc. which can provide competive pricing compared to Azure and GCP for certain types of resources. 
+
+This is an avenue we haven't fully investigated. In general, the service offerings are much smaller, so it is likely we'd need to use a combination of services. 
+
+**Risks**  
+1. More difficult to manage access control, onboarding/offboarding and maintining account access
+2. More things to document
+4. Typically less stable services (although there may be exceptions)
+5. Harder to monitor costs
+
+**Benefits**  
+1. Lower costs for certain resouce types
+
+## Assessment
+Intuitively an organization of our size and complexity should only use **one** cloud. However, since we already are spread accross two clouds and a range of technologies, the best solution becomes somewhat less clear cut.
+
+Specifically for end-user / high volume web applications, GCP is an appealing platform due to it's robust and secure nature. In addition, having applications that "talk" together in the same cloud is good for tracing, latency etc. 
+
+In general, the knowlege gained in one cloud, can be transferred to another (at a high level). Keeping the same "types" of applications in the same cloud could be beneficial for reuse of more specific knowledge.
+
+In addition, synchronizing users and groups from Azure to GCP has proved to work nicely so far.
+
+For specific applications (such as self-service BI), Azure is probably still a better platform whereas for some niche services (like streaming) other platforms (AWS) are more suitable.
+
+The recommendation for "trial" is to make GCP the "default" cloud for container-based web apps, and work on:
+1. Developing best practices for monitoring and tracing.
```

Comment:
Does this mean planning to move over our Application Insights setups. I know that Jakub has setup Google Trace for one of our Go APIs. Which could be a start.

Reply:
Yes. I just changed the recommendation to try both clouds though. So we should set up a .Net project with Google Trace and run a Go project on Azure

Reply:
@StevenMalaihollo